### DISCUSSION TITLE
Invalid Date

### DISCUSSION THREAD

**Discussion by Jckpt** [pocketbase/pocketbase#3644] [Q&A]
I'm trying to do this in my main.pb.js file:
const myDate = new Date("2023-10-30 14:00:00.000Z"); // the date is something my database return
console.log(myDate);

expected: Mon Oct 30 2023 14:00:00 GMT+0100 (CET)
got: Invalid Date

but when deleted the ".000Z" part from the string it works as intended

am I doing something wrong?

**Comment by ganigeorgiev** [comment]
~The above date format should work fine in Chrome and Firefox, but on Safari users reported in the past that it is a little more strict and throw an error on unsupported date formats.~
_On second read you are testing with the JS hooks, but the above is still true. In your case I guess the goja `Date` implementation follows also strictly the ES5 spec similar to Safari._

We use rfc3339 as a date format for backward compatibility with earlier PocketBase versions (_eventually in the future we'll introduce auto dates conversion for the filters and will change it to the more common one iso8601_).

As a workaround you can try replacing the space in the string with `T`, aka.:

```js
const myDate = new Date("2023-10-30 14:00:00.000Z".replace(" ", "T"));
```

**Reply by beardsleym** [reply]
Do we need to go the other way when saving to the db as well?

```js
const now = new Date()
now.toISOString() // '2023-11-18T19:35:54.645Z'
now.replace("T", " ")
```

**Reply by ganigeorgiev** [reply]
When persisting the date value it should work even if you send the ISO8601 string format, because we will try to cast it to the correct type (_based on the following [timeFormats map](https://github.com/spf13/cast/blob/e9ba3ce83919192b29c67da5bec158ce024fdcdb/caste.go#L1407-L1435)_), but in the database the value will be always stored in the above RFC3339 format.

**Reply by beardsleym** [reply]
I think I've found that I need to do  this for the filters though, which I guess makes sense as it's just a string.

```ts
const now = new Date()
now.toISOString() // '2023-11-18T19:35:54.645Z'
now.replace("T", " ")
```
